codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoder.java

```diff
@@ -101,4 +176,13 @@ protected void encodeName(String name, ByteBuf buf) throws Exception {
 
         buf.writeByte(0); // marks end of name field
     }
+
+    // Package private so it can be reused in the test.
+    static byte padWithZeros(byte b, int leftOverBits) {
+        int lastBit = Byte.SIZE - leftOverBits;
```

Comment:
The use case is very contained for this ... i think we can avoid all the conditionals and aritmatic here:
```java
        switch (leftOverBits) {
            case 8: return 0;
            case 7: return (byte) (0x01 & b);
            case 6: return (byte) (0x03 & b);
            case 5: return (byte) (0x07 & b);
            case 4: return (byte) (0x0F & b);
            case 3: return (byte) (0x1F & b);
            case 2: return (byte) (0x3F & b);
            case 1: return (byte) (0x7F & b);
            case 0: return b;
            default: throw new IllegalArgumentException("leftOverBits: " + leftOverBits);
        }
```